File: backend/app/schemas/analysis_job.py
"""Analysis job (async risk/lifestyle/full) document schema."""
from datetime import datetime, timezone
from typing import Optional

from pydantic import BaseModel, Field

from .enums import AnalysisJobStatus, AnalysisJobType
from .objectid import PyObjectId


def _utc_now() -> datetime:
    return datetime.now(timezone.utc)


class AnalysisJobDoc(BaseModel):
    """Mongo document shape for analysis_jobs collection. Tracks queued/running/completed jobs."""

    id: Optional[PyObjectId] = Field(None, alias="_id")
    patient_id: PyObjectId
    job_type: AnalysisJobType
    status: AnalysisJobStatus
    created_at: datetime = Field(default_factory=_utc_now)
    completed_at: Optional[datetime] = None
    error_message: Optional[str] = None

    model_config = {"populate_by_name": True, "arbitrary_types_allowed": True}

File: express-backend/src/models/AnalysisJob.ts
import mongoose, { Schema, type Document, type Model } from 'mongoose';

export interface AnalysisJobDocument extends Document {
  patient_id: mongoose.Types.ObjectId;
  job_type: string;
  status: string;
  created_at: Date;
  completed_at?: Date | null;
  error_message?: string | null;
}

const AnalysisJobSchema = new Schema<AnalysisJobDocument>(
  {
    patient_id: { type: Schema.Types.ObjectId, ref: 'Patient', required: true },
    job_type: { type: String, required: true },
    status: { type: String, required: true },
    created_at: { type: Date, default: () => new Date() },
    completed_at: { type: Date, default: null },
    error_message: { type: String, default: null },
  },
  {
    collection: 'analysis_jobs',
  },
);

export const AnalysisJob: Model<AnalysisJobDocument> =
  mongoose.models.AnalysisJob ||
  mongoose.model<AnalysisJobDocument>('AnalysisJob', AnalysisJobSchema);

